src/analyses/constant_propagator.cpp

```diff
@@ -204,19 +113,16 @@ void constant_propagator_domaint::transform(
     exprt g;
 
     if(from->get_target()==to)
-      g = simplify_expr(from->guard, ns);
+      g=simplify_expr(from->guard, ns);
     else
-      g = simplify_expr(not_exprt(from->guard), ns);
+      g=simplify_expr(not_exprt(from->guard), ns);
 
-    if (g.is_false())
+    if(g.is_false())
      values.set_to_bottom();
     else
     {
-      //TODO: we need to support widening!
-      if (g.is_constant())
-        values.set_to_top();
-      else
-        two_way_propagate_rec(g, ns);
+      two_way_propagate_rec(g, ns);
+      assert(!values.is_bottom); // for now
```

Comment:
Good question... @danpoe? @lucasccordeiro? I don't think I wrote this.

I can give a very meta-answer; this will be deprecated in favour of the variable precision domain configured to handle constants.